api/handler/acl.go

```diff
@@ -393,6 +420,14 @@ func (h *handler) PutObjectACLHandler(w http.ResponseWriter, r *http.Request) {
 		r.Header.Set(api.AmzACL, "")
 	}
 
+	if isBucketOwnerPreferred(eacl.EACL) {
+		if !isValidOwnerPreferred(r) {
+			h.logAndSendError(w, "header x-amz-acl:bucket-owner-full-control must be set", reqInfo, s3errors.GetAPIError(s3errors.ErrAccessDenied))
```

Comment:
Not sure it's a correct behavior for "preferred" case.

Reply:
Do you mean `PutObjectACLHandler` or such behavior in general?

Reply:
Requiring one specific ACL.